core/rawdb/freezer_table.go

```diff
@@ -1035,14 +1035,18 @@ func (t *freezerTable) resetItems(tail, head uint64) (*freezerTable, error) {
 	if t.readonly {
 		return nil, errors.New("resetItems in readonly mode")
 	}
+
+	t.lock.Lock()
```

Comment:
need lock here

Reply:
how about move the lock after the readonly check? so could have less lock/unlock

Reply:
`ResetTable()` will call `resetItems()`, then all the table's items will be removed? is it expected?